Makefile

```diff
@@ -1,13 +1,76 @@
-# Warning
+# This Makefile is (and must be) used by
+# travis/pre-commit.sh to qualify pull requests.
 #
-# This makefile is provided only as a convenience to run
-# the travis pre-commit script, which generates all the
-# code that needs to be generated, and runs all the tests.
+# That script generates all the code that needs
+# to be generated, and runs all the tests.
 #
-# This makefile won't be maintained until it replaces
-# (or is used by) the travis/pre-commit.sh script
+# Functionality in that script, expressed in bash, is
+# gradually being moved here.
 
+MYGOBIN := $(shell go env GOPATH)/bin
+
+.PHONY: all
 all:
 	./travis/pre-commit.sh
 
-.PHONY: all
+$(MYGOBIN)/golangci-lint:
```

Comment:
interesting.  do you want this in the target name?

Reply:
all i'd ever type is
```
make install-tools
```
or some such

Reply:
more directly answering question: yes.
someone might disagree.  another option would be to install all the dependent tools to some tmp dir and run from that.